### ISSUE TITLE
Table timemchine_modlog grows insanely

### ISSUE THREAD

**Issue by lab-at-nohl** [tine20/tine20#7398]
Maybe related to the modlog part of https://github.com/tine20/tine20/issues/6692 (not the xprops). 

My feeling is that it is related to LDAP again (openLDAP). I can date of increase to December 2020, so maybe a regression of the aforementioned? 

The suspicious entries amount on maybe 90 % of the table size (see below). Characteristics of the affected entries: 

- record_id = <setupuser>
- record_type = `Tinebase_Model_FullUser` (ca. 75 %) or `Tinebase_Model_User` (ca. 25 %)
- old_value = `NULL`
- new_value = JSON but binary unchanged compared against the seq before 

Conclusion/assumption: Somewhere during startup (setupuser) fetches NULL instead of FullUser or User. Thus, compared to the value it should be, a change is triggered... 

This is how I quantify the suspicious entries

```sql
SELECT count(*) FROM `tine20_timemachine_modlog` WHERE `modification_account` NOT LIKE '<setupuser>'

UNION 
SELECT count(*) FROM `tine20_timemachine_modlog` WHERE `modification_account` LIKE '<setupuser>'

UNION 
SELECT count(*) FROM `tine20_timemachine_modlog` WHERE `modification_account` LIKE '<setupuser>' AND `modification_time` > NOW() - INTERVAL 650 DAY 
```

In my DB this results to
- 10 % Changes not initiated by setupuser
- 90 % Changes by setupuser
- Nearly all of setupuser's changes (millions or GBytes!!) except a few thousand are max. 650 days old

I will try to provide a log, however, I have no clue how to filter atm.

**Comment by lab-at-nohl** [comment]
Short: The cause are missing RDN in LDAP, for which `Tinebase_User->_checkAndUpdateCurrentUser()` triggered a log of modification on _each_ call or at least each authentication. As such RDN are **not required but optional**, I would call this a bug. 

Longer explanation: During startup the `setupuser` compares LDAP against Tine 2.0 internal SQL DB (synchronization). If in LDAP a field is missing or empty, which PHP does not distinct, internal SQL DB will not be updated (i.e. set to empty string), which is good behavior (why lose the better information in Tine 2.0). Unfortunately an update (change) is detected though and written to modlog. Thus, the missing field causes a cycle in which every sync is supposed to be a change. The function can cope with non-existing fields, however, they have to be in `array $nonEmptyFields` which is misleading - by the name one would expect those fields to be synced even if they are empty. Finally, the optional RDN of LDAP are not registered here, only a very unusual one. 

A better, more transparent mechanism is needed. 

Also, wrongful entries in `timemachine_modlog` should be removed. I cannot provide a script to clean `timemachine_modlog` because I do not see how to find such entries definetly. But I can provide their characteristic: 

- `record_type` is either `Tinebase_Model_User` or `Tinebase_Model_FullUser`
- `modification_account` is setupuser not admin or similar
- `old_value` is `NULL` which is a bug by itself anyway
- `new_value` is always the same due to the unchanged diff between internal SQL DB and LDAP

I ended up with `DELETE FROM tine20_timemachine_modlog WHERE (record_type LIKE 'Tinebase_Model_User' OR record_type LIKE 'Tinebase_Model_FullUser') AND (client LIKE 'ActiveSync_%' OR client LIKE 'Tinebase_Server_WebDAV%');` because I did not see a scenario where a User should have been changed by this. It freed several GiB. But this is not the resolution for everyone because if LDAP would be changed out of Tine 2.0 indeed, the sync must have been done also by ActiveSync or WebDAV...
